crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -1460,32 +1569,51 @@ impl actor::HcP2p for HolochainP2pActor {
             let space_id = dna_hash.to_k2_space();
             let space = self.kitsune.space(space_id.clone()).await?;
             let loc = link_key.base.get_loc();
+            let agents = self
+                .get_random_peers_for_location("get_links", &space, loc)
+                .await?;
 
-            let (to_agent, to_url) = self.get_peer_for_loc("get_links", &space, loc).await?;
-
-            let r_options: event::GetLinksOptions = (&options).into();
+            let start = std::time::Instant::now();
 
-            let (msg_id, req) =
-                crate::wire::WireMessage::get_links_req(to_agent, link_key, r_options);
+            let out = select_ok_none_empty(
+                agents.into_iter().map(|(to_agent, to_url)| {
+                    Box::pin(async {
+                        let r_options: event::GetLinksOptions = (&options).into();
 
-            let start = std::time::Instant::now();
+                        let (msg_id, req) = crate::wire::WireMessage::get_links_req(
+                            to_agent,
+                            link_key.clone(),
+                            r_options,
+                        );
 
-            let out =
-                self.send_request("get_links", &space, to_url, msg_id, req, dna_hash, |res| {
-                    match res {
-                        crate::wire::WireMessage::GetLinksRes { response, .. } => {
-                            Ok(vec![response])
-                        }
-                        _ => Err(HolochainP2pError::other(format!(
-                            "invalid response to get_links: {res:?}"
-                        ))),
-                    }
-                })
-                .await;
+                        self.send_request(
+                            "get_links",
+                            &space,
+                            to_url,
+                            msg_id,
+                            req,
+                            dna_hash.clone(),
+                            |res| match res {
+                                crate::wire::WireMessage::GetLinksRes { response, .. } => {
+                                    Ok(response)
+                                }
+                                _ => Err(HolochainP2pError::other(format!(
+                                    "invalid response to get_links: {res:?}"
+                                ))),
+                            },
+                        )
+                        .await
+                    })
+                }),
+                |wire_link_ops| {
+                    wire_link_ops.creates.is_empty() && wire_link_ops.deletes.is_empty()
```

Comment:
Can there be a delete without a create?
```suggestion
                    wire_link_ops.creates.is_empty()
```